### DISCUSSION TITLE
Public repo can't access preset in private repo

### DISCUSSION THREAD

**Discussion by Jamesking56** [renovatebot/renovate#14866] [Request Help]
Hi,

My public dockerfiles repo can't seem to access our private org renovate configuration:

https://github.com/momentumstudioltd/dockerfiles/pull/13

You can see in the description there it says it can't access it.

Any idea why or how to give it access?

**Comment by rarkins** [comment]
This is intended behaviour and it's not currently configurable. We intentionally don't want a token with full read/write private privileges potentially leaking into a public repo, so we generate a token scoped to the current public repo only.

**Reply by Jamesking56** [reply]
What do you recommend we do in this situation?

**Reply by rarkins** [reply]
Does your public repo need access to secrets within your preset?

**Reply by Jamesking56** [reply]
No, just wanted the same settings but not have to duplicate them

**Reply by rarkins** [reply]
In this case I'd recommend something like this:
- Public repo containing the settings applicable to public repos. Ideally if your config is not secret then you could put nearly all the config into there
- Private repo (could be your existing one) containing the private or confidential settings

Then you have a choice:
1. Set your private preset to extend the public one, so that it contains both settings. Then have your private repos extend the private preset and the public repos extend the public preset, OR
2. Have all repos extend the public preset by default, and then add the private one to private repos (so that they extend 2 presets instead of 1). The private preset in this case does *not* need to extend the public

**Comment by enys** [comment]
In the case where it does indeed need access to secrets in the preset, such as access to private registry. What would be the most idiomatic way, without repeating secrets in each public repo ?

**Reply by rarkins** [reply]
Have a public preset with encrypted secrets and have repos extend that.
